### PR TITLE
put phone number unit test and refactor

### PR DESCRIPTION
**Pull request by jjames521**
Story: https://qmacbis.atlassian.net/browse/OY2-14301

### Details

Unit test for putPhoneNumber

### Changes

- Try catch block in putPhoneNumber to refactor
- Unit test add

### Implementation Notes

- Developer focused info that may affect future work

### Test Plan

1. Test step 1
2. Test step 2

### CODE REVIEW

File: services/app-api/putPhoneNumber.test.js [hunk 1]
```diff
@@ -1,8 +1,103 @@
...
+    id: "statesubmitteractive@cms.hhs.local",
+    statusCode: 200,
+    body: RESPONSE_CODE.VALIDATION_ERROR,
+    headers: {
+      "Access-Control-Allow-Origin": "*",
+      "Access-Control-Allow-Credentials": true,
+    },
+  };
+
+  await expect(main(mockObject))
```
Comment (kristin-at-theta) [inline services/app-api/putPhoneNumber.test.js:32]: await should no longer be needed, please remove :)

File: services/app-api/putPhoneNumber.test.js [hunk 2]
```diff
@@ -1,8 +1,103 @@
...
+    .resolves.toStrictEqual(expectedResponse)
+    .catch((e) => {
+      RESPONSE_CODE.VALIDATION_ERROR, e;
```
Comment (kristin-at-theta) [inline services/app-api/putPhoneNumber.test.js:35]: I think we want to console.log all testing errors, so please make this catch block the same as the final one: 
    .catch((e) => {
      console.log("error", e);
    });

**Review by kristin-at-theta — CHANGES_REQUESTED** [review]
I know this was a lot of work and it is nicely done.

I do think the final implementation could benefit from some cleanup :) 

changes requested: 

- mockObject and expectedResponse are duplicated in every test, can make them global and edit the differences found in each test.
- the await expect should no longer be needed now that you are mocking the dynamodb function
- I like that you catch the test errors, but I feel like they need to be logged as well.

These are minor updates that I believe will enhance the value of the code, but if we are running into timeline issues (or you just want to be done with it), none of my requests are showstoppers and ask and I will approve :)

File: services/app-api/putPhoneNumber.js [hunk 3]
```diff
@@ -29,9 +29,14 @@ const validateInput = (input) => {
  */
 export const main = handler(async (event) => {
   let input = event.body;
-  if (!isObject(input)) input = JSON.parse(input);
+  try {
+    if (!isObject(input)) input = JSON.parse(input);
```
[the remark this answers is not in the document]
Comment (jjames521) [reply services/app-api/putPhoneNumber.js:33]: Thank you!

File: services/app-api/putPhoneNumber.test.js (hunk 2 above)
Comment (jjames521) [reply services/app-api/putPhoneNumber.test.js]: Done!

**Review by g-s-k — APPROVED** [review]
looks good!

**Pull request merged**